pycroft/lib/finance.py

```diff
@@ -776,16 +776,77 @@ def match_activities():
 
     return matching
 
+
 @with_transaction
 def transaction_delete(transaction):
     if transaction.confirmed:
         raise ValueError("transaction already confirmed")
 
     session.session.delete(transaction)
 
+
 @with_transaction
 def transaction_confirm(transaction):
     if transaction.confirmed:
         raise ValueError("transaction already confirmed")
 
     transaction.confirmed = True
+
+
+def estimate_balance(user, end_date):
+    """
+    :param user: The member
+    :param end_date: Date of the end of the membership
+    :return: Estimated balance at the end_date
+    """
+
+    now = session.utcnow().date() + timedelta(1)
+
+    last_fee = MembershipFee.q.order_by(MembershipFee.ends_on.desc()).first()
+
+    if last_fee is None:
+        raise ValueError("no fee information available")
+
+    end_date_justified = end_date - timedelta(last_fee.booking_begin.days - 1)
+
+    months_to_pay = diff_month(end_date_justified, now)
+
+    # If the user has to pay a fee for the current month
+    if user.has_property('membership_fee',
+                         single(now.replace(day=last_fee.booking_end.days))):
+        months_to_pay += 1
+
+    last_month_last = now.replace(day=1) - timedelta(1)
+
+    split_last_month = (Split.q
+                        .filter_by(account=user.account)
+                        .join(Transaction)
+                        .filter(Split.amount > 0)
+                        .filter(Transaction.valid_on == last_month_last)
+                        ).first()
+
+    # If there was no fee booked yet for the last month and the user has to pay
+    # a fee for the last month, increment months_to_pay
+    if split_last_month is None:
+        if user.has_property('membership_fee',
+                             single(last_month_last
+                                    .replace(day=last_fee.booking_end.days))):
+            months_to_pay += 1
+
+    # Calculate last day of the current month
+    current_month_last = last_day_of_month(now)
+
+    # Get the split for a maybe existent membership fee booking in the current
+    # month
+    split_current_month = (Split.q
+                           .filter_by(account=user.account)
+                           .join(Transaction)
+                           .filter(Split.amount > 0)
+                           .filter(Transaction.valid_on == current_month_last)
+                           ).first()
```

Comment:
Might be extracted, is precisely the same as `split_last_month` modulovariation by `current_month_last`.